[Build Speed] Reduce includes of JSDOMPromiseDeferred.h and make it cheaper

jernoble opened the pull request:
#### 9784033bdce3c771fc8e88735612c51c396d423b
<pre>
[Build Speed] Reduce includes of JSDOMPromiseDeferred.h and make it cheaper
<a href="https://rdar.apple.com/164123237">rdar://164123237</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=302045">https://bugs.webkit.org/show_bug.cgi?id=302045</a>

Reviewed by Brent Fulgham and BJ Burg.

Reduce the number of times JSDOMPromiseDeferred.h is included during a build, and for those times
when including that header is unavoidable, reduce the cost of including that header.

For the first part, remove JSDOMPromiseDeferred.h from (almost all) non-Inlines.h headers by
converting those headers to use JSDOMPromiseDeferredForward.h. This also requires any class that
stores a JSDOMPromiseDeferred to use a std::unique_ptr instead of a std::optional. It also required
some classes to have inline methods move to the implementation file (or have implementation files
added where none previously existed). These include SimpleWritableStreamSink, ModuleScriptLoader,
NavigationAPIMethodTracker, NavigationTransition, and WorkletPendingTasks.

For the second part, remove JSDOMConvert.h from JSDOMPromiseDeferred.h. JSDOMConvert.h pulls in
every different mechanism to convert a c++ object to an IDLType, which is very expensive. Instead,
the users of JSDOMPromiseDeferred.h should include only the specific JSDOMConvert*.h header needed
to call resolve() or reject().

Before this change, JSDOMPromiseDeferred.h was the 11th most expensive header in the WebCore Unified
build. It was included 184 times, at an average cost on this machine of 880ms of CPU time, for a total
cost of 2.7m of CPU time spent parsing this header.

After this change, JSDOMPromiseDeferred.h is the 67th most expensive header in the WebCore Unified
build. It is included 120 times, at an average cost on this machine of 430ms of CPU time, for a total
cost of 53s of CPU time spent parsing this header.

All told, frontend parsing time decreased from 8122s to 7876s, a reduction of 3%.

* Source/WebCore/Modules/WebGPU/GPUBuffer.cpp:
(WebCore::GPUBuffer::mapAsync):
(WebCore::GPUBuffer::internalUnmap):
* Source/WebCore/Modules/WebGPU/GPUBuffer.h:
* Source/WebCore/Modules/WebGPU/GPUQueue.h:
* Source/WebCore/Modules/WebGPU/GPUShaderModule.cpp:
* Source/WebCore/Modules/applepay/ApplePaySession.cpp:
* Source/WebCore/Modules/applepay/ApplePaySetup.cpp:
(WebCore::ApplePaySetup::getSetupFeatures):
(WebCore::ApplePaySetup::begin):
(WebCore::ApplePaySetup::stop):
* Source/WebCore/Modules/applepay/ApplePaySetupWebCore.h:
* Source/WebCore/Modules/cache/DOMCache.cpp:
* Source/WebCore/Modules/cache/DOMCacheStorage.cpp:
* Source/WebCore/Modules/contact-picker/ContactsManager.cpp:
* Source/WebCore/Modules/cookie-consent/NavigatorCookieConsent.cpp:
* Source/WebCore/Modules/cookie-store/CookieStore.cpp:
* Source/WebCore/Modules/credentialmanagement/CredentialsContainer.cpp:
* Source/WebCore/Modules/encryptedmedia/MediaKeySession.cpp:
* Source/WebCore/Modules/encryptedmedia/MediaKeySystemAccess.cpp:
* Source/WebCore/Modules/encryptedmedia/MediaKeys.cpp:
* Source/WebCore/Modules/encryptedmedia/NavigatorEME.cpp:
* Source/WebCore/Modules/fetch/FetchBody.cpp:
* Source/WebCore/Modules/fetch/FetchBodyConsumer.cpp:
* Source/WebCore/Modules/fetch/WindowOrWorkerGlobalScopeFetch.cpp:
* Source/WebCore/Modules/filesystem/FileSystemDirectoryHandle.cpp:
* Source/WebCore/Modules/filesystem/FileSystemWritableFileStreamSink.cpp:
* Source/WebCore/Modules/gamepad/GamepadHapticActuator.cpp:
* Source/WebCore/Modules/identity/CredentialRequestCoordinator.cpp:
(WebCore::CredentialRequestCoordinator::setCurrentPromise):
(WebCore::CredentialRequestCoordinator::currentPromise):
* Source/WebCore/Modules/identity/CredentialRequestCoordinator.h:
* Source/WebCore/Modules/identity/DigitalCredential.cpp:
* Source/WebCore/Modules/identity/DigitalCredential.h:
* Source/WebCore/Modules/indexeddb/IDBFactory.cpp:
* Source/WebCore/Modules/mediastream/ImageCapture.cpp:
* Source/WebCore/Modules/mediastream/ImageCapture.h:
* Source/WebCore/Modules/mediastream/MediaDevices.cpp:
* Source/WebCore/Modules/mediastream/MediaStreamTrack.cpp:
* Source/WebCore/Modules/mediastream/MediaStreamTrack.h:
* Source/WebCore/Modules/mediastream/PeerConnectionBackend.cpp:
* Source/WebCore/Modules/mediastream/RTCEncodedStreamProducer.cpp:
* Source/WebCore/Modules/mediastream/RTCRtpSFrameTransform.cpp:
* Source/WebCore/Modules/mediastream/RTCRtpScriptTransformer.cpp:
* Source/WebCore/Modules/mediastream/UserMediaRequest.cpp:
* Source/WebCore/Modules/mediastream/VideoTrackGenerator.cpp:
* Source/WebCore/Modules/mediastream/VideoTrackGenerator.h:
* Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCMediaEndpoint.cpp:
* Source/WebCore/Modules/notifications/Notification.cpp:
* Source/WebCore/Modules/paymentrequest/PaymentRequest.cpp:
* Source/WebCore/Modules/permissions/Permissions.cpp:
* Source/WebCore/Modules/pictureinpicture/DocumentPictureInPicture.cpp:
* Source/WebCore/Modules/pictureinpicture/HTMLVideoElementPictureInPicture.cpp:
* Source/WebCore/Modules/push-api/PushSubscription.cpp:
* Source/WebCore/Modules/remoteplayback/RemotePlayback.cpp:
* Source/WebCore/Modules/screen-wake-lock/WakeLock.cpp:
* Source/WebCore/Modules/storage/StorageManager.cpp:
* Source/WebCore/Modules/streams/StreamPipeToUtilities.cpp:
* Source/WebCore/Modules/streams/WritableStreamSink.cpp: Copied from Source/WebCore/Modules/WebGPU/GPUShaderModule.cpp.
(WebCore::SimpleWritableStreamSink::SimpleWritableStreamSink):
(WebCore::SimpleWritableStreamSink::write):
* Source/WebCore/Modules/streams/WritableStreamSink.h:
(WebCore::SimpleWritableStreamSink::SimpleWritableStreamSink): Deleted.
(WebCore::SimpleWritableStreamSink::write): Deleted.
* Source/WebCore/Modules/web-locks/WebLockManager.cpp:
* Source/WebCore/Modules/webaudio/OfflineAudioContext.cpp:
* Source/WebCore/Modules/webauthn/AuthenticatorCoordinator.cpp:
* Source/WebCore/Modules/webcodecs/WebCodecsVideoFrame.cpp:
* Source/WebCore/Modules/webtransport/DatagramSink.cpp:
* Source/WebCore/Modules/webtransport/WebTransport.cpp:
* Source/WebCore/Modules/webtransport/WebTransportSendStreamSink.cpp:
* Source/WebCore/Modules/webtransport/WorkerWebTransportSession.cpp:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/animation/WebAnimation.cpp:
* Source/WebCore/bindings/js/DOMPromiseProxy.h:
* Source/WebCore/bindings/js/JSDOMConvertEnumeration.h:
* Source/WebCore/bindings/js/JSDOMGuardedObject.h:
* Source/WebCore/bindings/js/JSDOMPromiseDeferred.h:
* Source/WebCore/bindings/js/ModuleScriptLoader.cpp: Copied from Source/WebCore/Modules/WebGPU/GPUShaderModule.cpp.
(WebCore::ModuleScriptLoader::ModuleScriptLoader):
* Source/WebCore/bindings/js/ModuleScriptLoader.h:
(WebCore::ModuleScriptLoader::ModuleScriptLoader): Deleted.
* Source/WebCore/crypto/SubtleCrypto.cpp:
* Source/WebCore/css/DOMCSSPaintWorklet.cpp:
(WebCore::PaintWorklet::addModule):
* Source/WebCore/css/DOMCSSPaintWorklet.h:
* Source/WebCore/css/FontFace.cpp:
* Source/WebCore/css/FontFaceSet.cpp:
* Source/WebCore/dom/DocumentStorageAccess.cpp:
* Source/WebCore/dom/InternalObserverEvery.cpp:
* Source/WebCore/dom/InternalObserverFind.cpp:
* Source/WebCore/dom/InternalObserverFirst.cpp:
* Source/WebCore/dom/InternalObserverForEach.cpp:
* Source/WebCore/dom/InternalObserverSome.cpp:
* Source/WebCore/fileapi/Blob.cpp:
* Source/WebCore/html/OffscreenCanvas.cpp:
* Source/WebCore/inspector/InspectorFrontendHost.cpp:
* Source/WebCore/page/Navigation.cpp:
(WebCore::NavigationAPIMethodTracker::NavigationAPIMethodTracker):
* Source/WebCore/page/Navigation.h:
(WebCore::NavigationAPIMethodTracker::NavigationAPIMethodTracker): Deleted.
* Source/WebCore/page/NavigationTransition.cpp:
(WebCore::NavigationTransition::create):
* Source/WebCore/page/NavigationTransition.h:
* Source/WebCore/page/NavigatorLoginStatus.cpp:
* Source/WebCore/page/NavigatorUAData.cpp:
(WebCore::NavigatorUAData::createArbitraryVersion): Deleted.
(WebCore::NavigatorUAData::createArbitraryBrand): Deleted.
* Source/WebCore/page/NavigatorUAData.h:
(): Deleted.
* Source/WebCore/page/Quirks.cpp:
(WebCore::handleVimeoQuirks):
* Source/WebCore/page/UserMessageHandler.cpp:
* Source/WebCore/testing/ServiceWorkerInternals.cpp:
* Source/WebCore/workers/service/ServiceWorkerClients.cpp:
* Source/WebCore/workers/service/ServiceWorkerContainer.cpp:
* Source/WebCore/workers/service/ServiceWorkerWindowClient.cpp:
* Source/WebCore/workers/service/background-fetch/BackgroundFetchManager.cpp:
* Source/WebCore/workers/service/background-fetch/BackgroundFetchManager.h:
* Source/WebCore/workers/service/background-fetch/BackgroundFetchRecord.cpp:
* Source/WebCore/workers/service/background-fetch/BackgroundFetchRegistration.cpp:
* Source/WebCore/workers/service/background-fetch/BackgroundFetchRegistration.h:
* Source/WebCore/workers/service/background-fetch/BackgroundFetchUpdateUIEvent.h:
* Source/WebCore/worklets/Worklet.cpp:
(WebCore::Worklet::addModule):
* Source/WebCore/worklets/Worklet.h:
* Source/WebCore/worklets/WorkletPendingTasks.cpp:
(WebCore::WorkletPendingTasks::create):
(WebCore::WorkletPendingTasks::WorkletPendingTasks):
(WebCore::WorkletPendingTasks::abort):
(WebCore::WorkletPendingTasks::decrementCounter):
* Source/WebCore/worklets/WorkletPendingTasks.h:
(WebCore::WorkletPendingTasks::create): Deleted.

Canonical link: <a href="https://commits.webkit.org/309275@main">https://commits.webkit.org/309275@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8a10d211a6a5c25d4b50cf175af3ded96a965a35

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/150112 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/22870 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/168/builds/16431 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/158824 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/103546 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/5ca2942b-6f77-4074-b4c3-ed2c23e633ff) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/151985 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/23300 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/22970 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/115799 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/82258 "3 flakes 1 failures") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/b1cfca7e-5ddc-44b4-9dbc-6ea263a98a6c) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/153072 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/17912 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/134676 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/96528 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/1278641e-7706-475e-9bd7-6d04f60d1342) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/17012 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/14961 "Passed tests") | [✅ 🛠 gtk3-libwebrtc](https://ews-build.webkit.org/#/builders/173/builds/6669 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/126627 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/167/builds/12600 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/161297 "Built successfully") | | 
| | [✅ 🛠 ios-safer-cpp](https://ews-build.webkit.org/#/builders/174/builds/4388 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/169/builds/14152 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/123803 "Passed tests") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/22672 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/19008 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/124004 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/33677 "Built successfully and passed tests") | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/22659 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/134395 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/78888 "Built successfully") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/19131 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/170/builds/11152 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/22272 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/86072 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/21986 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/22138 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/22040 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Source/WebCore/page/NavigatorUAData.cpp:
```diff
@@ -90,35 +96,27 @@ Ref<NavigatorUAData> NavigatorUAData::create(Ref<UserAgentStringData>&& userAgen
-        std::ranges::shuffle(temp, rng);
-        brandVersion.construct(temp);
-    });
+    static NeverDestroyed<Vector<NavigatorUABrandVersion>> brandVersion = createBrandFromBrowserNameAndVersion("AppleWebKit"_s, "605.1.15"_s);
```
This all seems much cleaner, and I believe it matches behavior with the old code.

brentfulgham reviewed (approved):
I think the NavigatorUAData changes look correct (@colelao can you please double-check)?

The other files looked right (simple changes).

r=mews

colelao commented on Source/WebCore/page/NavigatorUAData.cpp:
```diff
@@ -163,7 +161,7 @@ UALowEntropyJSON NavigatorUAData::toJSON() const
 {
     auto values = UADataValues::create(brands(), mobile(), platform());
-    if (overrideFromUserAgentString) {
+    if (m_platformOverride) [[unlikely]] {
```
I believe this should be
```
if (m_platformOverride || m_mobileOverride || m_brandOverride) [[unlikely]]
```

The user agent string can be overridden, but there is no guarantee there will be a platform present in the string provided.

jernoble committed:
[Build Speed] Reduce includes of JSDOMPromiseDeferred.h and make it cheaper
rdar://164123237
https://bugs.webkit.org/show_bug.cgi?id=302045

Reviewed by Brent Fulgham and BJ Burg.

Reduce the number of times JSDOMPromiseDeferred.h is included during a build, and for those times
when including that header is unavoidable, reduce the cost of including that header.

For the first part, remove JSDOMPromiseDeferred.h from (almost all) non-Inlines.h headers by
converting those headers to use JSDOMPromiseDeferredForward.h. This also requires any class that
stores a JSDOMPromiseDeferred to use a std::unique_ptr instead of a std::optional. It also required
some classes to have inline methods move to the implementation file (or have implementation files
added where none previously existed). These include SimpleWritableStreamSink, ModuleScriptLoader,
NavigationAPIMethodTracker, NavigationTransition, and WorkletPendingTasks.

For the second part, remove JSDOMConvert.h from JSDOMPromiseDeferred.h. JSDOMConvert.h pulls in
every different mechanism to convert a c++ object to an IDLType, which is very expensive. Instead,
the users of JSDOMPromiseDeferred.h should include only the specific JSDOMConvert*.h header needed
to call resolve() or reject().

Before this change, JSDOMPromiseDeferred.h was the 11th most expensive header in the WebCore Unified
build. It was included 184 times, at an average cost on this machine of 880ms of CPU time, for a total
cost of 2.7m of CPU time spent parsing this header.

After this change, JSDOMPromiseDeferred.h is the 67th most expensive header in the WebCore Unified
build. It is included 120 times, at an average cost on this machine of 430ms of CPU time, for a total
cost of 53s of CPU time spent parsing this header.

All told, frontend parsing time decreased from 8122s to 7876s, a reduction of 3%.

* Source/WebCore/Modules/WebGPU/GPUBuffer.cpp:
(WebCore::GPUBuffer::mapAsync):
(WebCore::GPUBuffer::internalUnmap):
* Source/WebCore/Modules/WebGPU/GPUBuffer.h:
* Source/WebCore/Modules/WebGPU/GPUQueue.h:
* Source/WebCore/Modules/WebGPU/GPUShaderModule.cpp:
* Source/WebCore/Modules/applepay/ApplePaySession.cpp:
* Source/WebCore/Modules/applepay/ApplePaySetup.cpp:
(WebCore::ApplePaySetup::getSetupFeatures):
(WebCore::ApplePaySetup::begin):
(WebCore::ApplePaySetup::stop):
* Source/WebCore/Modules/applepay/ApplePaySetupWebCore.h:
* Source/WebCore/Modules/cache/DOMCache.cpp:
* Source/WebCore/Modules/cache/DOMCacheStorage.cpp:
* Source/WebCore/Modules/contact-picker/ContactsManager.cpp:
* Source/WebCore/Modules/cookie-consent/NavigatorCookieConsent.cpp:
* Source/WebCore/Modules/cookie-store/CookieStore.cpp:
* Source/WebCore/Modules/credentialmanagement/CredentialsContainer.cpp:
* Source/WebCore/Modules/encryptedmedia/MediaKeySession.cpp:
* Source/WebCore/Modules/encryptedmedia/MediaKeySystemAccess.cpp:
* Source/WebCore/Modules/encryptedmedia/MediaKeys.cpp:
* Source/WebCore/Modules/encryptedmedia/NavigatorEME.cpp:
* Source/WebCore/Modules/fetch/FetchBody.cpp:
* Source/WebCore/Modules/fetch/FetchBodyConsumer.cpp:
* Source/WebCore/Modules/fetch/WindowOrWorkerGlobalScopeFetch.cpp:
* Source/WebCore/Modules/filesystem/FileSystemDirectoryHandle.cpp:
* Source/WebCore/Modules/filesystem/FileSystemWritableFileStreamSink.cpp:
* Source/WebCore/Modules/gamepad/GamepadHapticActuator.cpp:
* Source/WebCore/Modules/identity/CredentialRequestCoordinator.cpp:
(WebCore::CredentialRequestCoordinator::setCurrentPromise):
(WebCore::CredentialRequestCoordinator::currentPromise):
* Source/WebCore/Modules/identity/CredentialRequestCoordinator.h:
* Source/WebCore/Modules/identity/DigitalCredential.cpp:
* Source/WebCore/Modules/identity/DigitalCredential.h:
* Source/WebCore/Modules/indexeddb/IDBFactory.cpp:
* Source/WebCore/Modules/mediastream/ImageCapture.cpp:
* Source/WebCore/Modules/mediastream/ImageCapture.h:
* Source/WebCore/Modules/mediastream/MediaDevices.cpp:
* Source/WebCore/Modules/mediastream/MediaStreamTrack.cpp:
* Source/WebCore/Modules/mediastream/MediaStreamTrack.h:
* Source/WebCore/Modules/mediastream/PeerConnectionBackend.cpp:
* Source/WebCore/Modules/mediastream/RTCEncodedStreamProducer.cpp:
* Source/WebCore/Modules/mediastream/RTCRtpSFrameTransform.cpp:
* Source/WebCore/Modules/mediastream/RTCRtpScriptTransformer.cpp:
* Source/WebCore/Modules/mediastream/UserMediaRequest.cpp:
* Source/WebCore/Modules/mediastream/VideoTrackGenerator.cpp:
* Source/WebCore/Modules/mediastream/VideoTrackGenerator.h:
* Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCMediaEndpoint.cpp:
* Source/WebCore/Modules/notifications/Notification.cpp:
* Source/WebCore/Modules/paymentrequest/PaymentRequest.cpp:
* Source/WebCore/Modules/permissions/Permissions.cpp:
* Source/WebCore/Modules/pictureinpicture/DocumentPictureInPicture.cpp:
* Source/WebCore/Modules/pictureinpicture/HTMLVideoElementPictureInPicture.cpp:
* Source/WebCore/Modules/push-api/PushSubscription.cpp:
* Source/WebCore/Modules/remoteplayback/RemotePlayback.cpp:
* Source/WebCore/Modules/screen-wake-lock/WakeLock.cpp:
* Source/WebCore/Modules/storage/StorageManager.cpp:
* Source/WebCore/Modules/streams/StreamPipeToUtilities.cpp:
* Source/WebCore/Modules/streams/WritableStreamSink.cpp: Copied from Source/WebCore/Modules/WebGPU/GPUShaderModule.cpp.
(WebCore::SimpleWritableStreamSink::SimpleWritableStreamSink):
(WebCore::SimpleWritableStreamSink::write):
* Source/WebCore/Modules/streams/WritableStreamSink.h:
(WebCore::SimpleWritableStreamSink::SimpleWritableStreamSink): Deleted.
(WebCore::SimpleWritableStreamSink::write): Deleted.
* Source/WebCore/Modules/web-locks/WebLockManager.cpp:
* Source/WebCore/Modules/webaudio/OfflineAudioContext.cpp:
* Source/WebCore/Modules/webauthn/AuthenticatorCoordinator.cpp:
* Source/WebCore/Modules/webcodecs/WebCodecsVideoFrame.cpp:
* Source/WebCore/Modules/webtransport/DatagramSink.cpp:
* Source/WebCore/Modules/webtransport/WebTransport.cpp:
* Source/WebCore/Modules/webtransport/WebTransportSendStreamSink.cpp:
* Source/WebCore/Modules/webtransport/WorkerWebTransportSession.cpp:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/animation/WebAnimation.cpp:
* Source/WebCore/bindings/js/DOMPromiseProxy.h:
* Source/WebCore/bindings/js/JSDOMConvertEnumeration.h:
* Source/WebCore/bindings/js/JSDOMGuardedObject.h:
* Source/WebCore/bindings/js/JSDOMPromiseDeferred.h:
* Source/WebCore/bindings/js/ModuleScriptLoader.cpp: Copied from Source/WebCore/Modules/WebGPU/GPUShaderModule.cpp.
(WebCore::ModuleScriptLoader::ModuleScriptLoader):
* Source/WebCore/bindings/js/ModuleScriptLoader.h:
(WebCore::ModuleScriptLoader::ModuleScriptLoader): Deleted.
* Source/WebCore/crypto/SubtleCrypto.cpp:
* Source/WebCore/css/DOMCSSPaintWorklet.cpp:
(WebCore::PaintWorklet::addModule):
* Source/WebCore/css/DOMCSSPaintWorklet.h:
* Source/WebCore/css/FontFace.cpp:
* Source/WebCore/css/FontFaceSet.cpp:
* Source/WebCore/dom/DocumentStorageAccess.cpp:
* Source/WebCore/dom/InternalObserverEvery.cpp:
* Source/WebCore/dom/InternalObserverFind.cpp:
* Source/WebCore/dom/InternalObserverFirst.cpp:
* Source/WebCore/dom/InternalObserverForEach.cpp:
* Source/WebCore/dom/InternalObserverSome.cpp:
* Source/WebCore/fileapi/Blob.cpp:
* Source/WebCore/html/OffscreenCanvas.cpp:
* Source/WebCore/inspector/InspectorFrontendHost.cpp:
* Source/WebCore/page/Navigation.cpp:
(WebCore::NavigationAPIMethodTracker::NavigationAPIMethodTracker):
* Source/WebCore/page/Navigation.h:
(WebCore::NavigationAPIMethodTracker::NavigationAPIMethodTracker): Deleted.
* Source/WebCore/page/NavigationTransition.cpp:
(WebCore::NavigationTransition::create):
* Source/WebCore/page/NavigationTransition.h:
* Source/WebCore/page/NavigatorLoginStatus.cpp:
* Source/WebCore/page/NavigatorUAData.cpp:
(WebCore::NavigatorUAData::createArbitraryVersion): Deleted.
(WebCore::NavigatorUAData::createArbitraryBrand): Deleted.
* Source/WebCore/page/NavigatorUAData.h:
(): Deleted.
* Source/WebCore/page/Quirks.cpp:
(WebCore::handleVimeoQuirks):
* Source/WebCore/page/UserMessageHandler.cpp:
* Source/WebCore/testing/ServiceWorkerInternals.cpp:
* Source/WebCore/workers/service/ServiceWorkerClients.cpp:
* Source/WebCore/workers/service/ServiceWorkerContainer.cpp:
* Source/WebCore/workers/service/ServiceWorkerWindowClient.cpp:
* Source/WebCore/workers/service/background-fetch/BackgroundFetchManager.cpp:
* Source/WebCore/workers/service/background-fetch/BackgroundFetchManager.h:
* Source/WebCore/workers/service/background-fetch/BackgroundFetchRecord.cpp:
* Source/WebCore/workers/service/background-fetch/BackgroundFetchRegistration.cpp:
* Source/WebCore/workers/service/background-fetch/BackgroundFetchRegistration.h:
* Source/WebCore/workers/service/background-fetch/BackgroundFetchUpdateUIEvent.h:
* Source/WebCore/worklets/Worklet.cpp:
(WebCore::Worklet::addModule):
* Source/WebCore/worklets/Worklet.h:
* Source/WebCore/worklets/WorkletPendingTasks.cpp:
(WebCore::WorkletPendingTasks::create):
(WebCore::WorkletPendingTasks::WorkletPendingTasks):
(WebCore::WorkletPendingTasks::abort):
(WebCore::WorkletPendingTasks::decrementCounter):
* Source/WebCore/worklets/WorkletPendingTasks.h:
(WebCore::WorkletPendingTasks::create): Deleted.

Canonical link: https://commits.webkit.org/309275@main
